src/Ship.js

```diff
@@ -0,0 +1,24 @@
+function Ship(itinerary){
+    this.itinerary = itinerary;
+    this.currentPort = itinerary.ports[0];
+    this.previousPort = null;
+    this.currentPort.addShip(this);
+    this.setSail = function(){
+        const itinerary = this.itinerary;
+        const currentPortIndex = itinerary.ports.indexOf(this.currentPort);
```

Comment:
Why are you creating a new variable and assigning a value to it that already exists? Could `this.itinerary` not just be used directly?
```suggestion
        const currentPortIndex = this.itinerary.ports.indexOf(this.currentPort);
```